app/src/main/java/com/aconno/sensorics/dagger/mainactivity/MainActivityModule.kt

```diff
@@ -105,11 +106,15 @@ class MainActivityModule {
         sensoricsApplication
     )
 
+
     @Provides
     @MainActivityScope
-    fun providePermissionsViewModel(mainActivity: MainActivity): PermissionViewModel {
+    fun providePermissionsHandler(mainActivity: MainActivity): SensoricsPermissionsHandler {
         val permissionAction = PermissionActionFactory.getPermissionAction(mainActivity)
-        return PermissionViewModel(permissionAction, mainActivity)
+        val permissionsHandler = SensoricsPermissionsHandler(mainActivity)
+        val permissionViewModel = PermissionViewModel(permissionAction, permissionsHandler)
+        permissionsHandler.permissionViewModel = permissionViewModel
```

Comment:
This is a red flag. We are doing something wrong if we need to have circular dependencies.